src/main/scala/util/SynchronizerReg.scala

```diff
@@ -101,13 +108,17 @@ object AsyncResetSynchronizerShiftReg {
     apply (in, sync, init.litValue.toInt, None)
 }
 
+// Note: This module may end up with a non-Bool type reset.
```

Comment:
hmm either this comment is wrong or the `with RequireSyncReset` should be removed